### PR TITLE
chore(mirrorbits): set images `pullPolicy` to `IfNotPresent`

### LINKED ISSUE #590: `mirrorbits` image pull policies should be set like every other to `IfNotPresent`
This is the only chart defining its `image.*.pullPolicy` to `Always`: https://github.com/jenkins-infra/helm-charts/blob/a345f10a82f9ec6ce5f81c5fdf1ebfb2dddb58f4/charts/mirrorbits/values.yaml#L11-L15

Each of our other charts are defining them to [`IfNotPresent`](https://github.com/search?q=repo%3Ajenkins-infra%2Fhelm-charts%20pullPolicy&type=code), and it forces us to override them in jenkins-infra/kubernetes-management:
https://github.com/jenkins-infra/kubernetes-management/blob/2a4c2d2724e72f198f9bad9da1a2fb271d27e23a/config/mirrorbits.yaml#L4-L9

### PR DESCRIPTION
**Pull request by lemeurherve**
Fixes #590

### CODE REVIEW

**Review by dduportal — APPROVED** [review]
LGTM!

- It make sense as it would be set to the default behavior of Kubernetes (https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting)

> if you omit the `imagePullPolicy` field, and you specify the digest for the container image, the `imagePullPolicy` is automatically set to `IfNotPresent`.

- Should not have any impact changing from `Always` to `IfNotPresent` when a digest is specified as per https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy:

> `Always`:
> every time the kubelet launches a container, the kubelet queries the container image registry to resolve the name to an image [digest](https://docs.docker.com/engine/reference/commandline/pull/#pull-an-image-by-digest-immutable-identifier). If the kubelet has a container image with that exact digest cached locally, the kubelet uses its cached image; otherwise, the kubelet pulls the image with the resolved digest, and uses that image to launch the container.

**Comment by dduportal** [comment]
In term of "pure" helm change, this change should be a minor change, not a patch though

File: charts/mirrorbits/tests/custom_values_test.yaml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 suite: Tests with custom values
+values:
+  - default_secrets_values.yaml
 templates:
   - service.files.yaml
+  - deployment.yaml
+  - secret.yaml # Direct dependency of deployment.yaml
+  - deployment.files.yaml
```
Comment (lemeurherve) [inline charts/mirrorbits/tests/custom_values_test.yaml:8]:
```suggestion
  - deployment.yaml
  - deployment.files.yaml
  - secret.yaml # Direct dependency of deployment.yaml
  - service.files.yaml
```
Nit, ordered.

File: charts/mirrorbits/tests/custom_values_test.yaml [hunk 2]
```diff
@@ -32,3 +37,31 @@ tests:
...
+          pullPolicy: Never
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Deployment
+      - equal:
+          path: "spec.template.spec.containers[*].imagePullPolicy"
+          value: Never
+  - it: Should define a customized "mirrorbits" deployment
```
Comment (lemeurherve) [inline charts/mirrorbits/tests/custom_values_test.yaml:54]:
```suggestion
  - it: should define a customized "files" deployment
    template: deployment.files.yaml
    set:
      image:
        files:
          pullPolicy: Never
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - equal:
          path: "spec.template.spec.containers[*].imagePullPolicy"
          value: Never
  - it: should define a customized "mirrorbits" deployment
```
Nit, same as the existing ones and as the doc examples from helm-unittest.

File: charts/mirrorbits/tests/defaults_test.yaml [hunk 3]
```diff
@@ -4,6 +4,8 @@ values:
   - default_secrets_values.yaml
 templates:
   - deployment.files.yaml
+  - deployment.yaml
+  - secret.yaml # Direct dependency of deployment.yaml
   - ingress.yaml
   - service.files.yaml
```
Comment (lemeurherve) [inline charts/mirrorbits/tests/defaults_test.yaml:10]:
```suggestion
  - deployment.files.yaml
  - deployment.yaml
  - ingress.yaml
  - secret.yaml # Direct dependency of deployment.yaml
  - service.files.yaml
```
Same as above.

File: charts/mirrorbits/tests/defaults_test.yaml [hunk 4]
```diff
@@ -25,7 +27,7 @@ tests:
       - equal:
           path: spec.selector["app.kubernetes.io/instance"]
           value: "RELEASE-NAME-files"
-  - it: Should set the correct deployment labels
+  - it: Should define the default "files" deployment
```
Comment (lemeurherve) [inline charts/mirrorbits/tests/defaults_test.yaml:30]:
```suggestion
  - it: should define the default "files" deployment
```
Same as above.

File: charts/mirrorbits/tests/defaults_test.yaml [hunk 5]
```diff
@@ -38,3 +40,16 @@ tests:
       - equal:
           path: spec.template.metadata.labels["app.kubernetes.io/instance"]
           value: "RELEASE-NAME-files"
+      - equal:
+          path: "spec.template.spec.containers[*].imagePullPolicy"
+          value: IfNotPresent
+  - it: Should define the default "mirrorbits" deployment
```
Comment (lemeurherve) [inline charts/mirrorbits/tests/defaults_test.yaml:46]:
```suggestion
  - it: should define the default "mirrorbits" deployment
```
Same as above.

**Review by dduportal — APPROVED** [review]

**Pull request merged**